### PR TITLE
add experimental features concepts doc

### LINKED ISSUE #192: create a document as the entry for experimental features
Since version `v1.0.0-rc.4` Notation started supporting experimental features. Users should set `NOTATION_EXPERIMENTAL=1` to use experimental features. Here is current feature list of rc.4

- Store signatures using artifact manifest (Require Registry support)
- Signing, listing and verifying images as OCI image layout

It is foreseen that more experimental features will be introduced in the future, and some experimental features could be completely removed as the nature of experimental over time. So, it will be convenient for users to know what experimental features that notation supported, and how to use them. 

The proposal is that we can create a document which lists all the experimental features that the latest Notation supports, and provides guideline links to those features if they are available.

### PR DESCRIPTION
**Pull request by zr-msft**
fixes #192

### CODE REVIEW

File: content/en/docs/concepts/experimental-features.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+type: docs
+weight: 10
+---
+
+Starting in version `v1.0.0-rc.4`, Notation began introducing experimental features. Experimental features are features that are not yet ready for production use but are available for testing and evaluation. Experimental features may change significantly in later releases or be completely removed. Experimental features are not enabled by default and are not supported.
+
+Currently, Notation has the following experimental features:
+
+- Signing, listing and verifying images as OCI image layout
+- Store signatures using artifact manifest (requires registry support)
```
Comment (yizha1) [inline content/en/docs/concepts/experimental-features.md:13]: We will soon release RC-5 this week. The second experimental feature will be removed, and a new one will be introduced. I will provide more details once the implementation is done.

**Review by yizha1 — COMMENTED** [review]
Thanks @zr-msft

File: content/en/docs/concepts/experimental-features.md (hunk 1 above)
Comment (zr-msft) [reply content/en/docs/concepts/experimental-features.md:13]: Thank you @yizha1 

Will also add a link once #196 is merged

Comment (yizha1) [reply content/en/docs/concepts/experimental-features.md:13]: @zr-msft RC-5 was released, however the blog PR still not merged yet. In RC-5, new flag was introduced named `--allow-referrers-api`, see https://deploy-preview-218--notarydev.netlify.app/blog/2023/announcing-notation-rc5/#change-default-behavior-and-introduce-a-new-flag---allow-referrers-api for details. 

This is the flag that was deprecated in RC-5: https://deploy-preview-218--notarydev.netlify.app/blog/2023/announcing-notation-rc5/#deprecate-flag---signature-manifest-in-notation-sign

Comment (zr-msft) [reply content/en/docs/concepts/experimental-features.md]: Thanks @yizha1 amended PR to include your feedback

File: content/en/docs/concepts/experimental-features.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+weight: 10
+---
+
+Starting in version `v1.0.0-rc.4`, Notation began introducing experimental features. Experimental features are features that are not yet ready for production use but are available for testing and evaluation. Experimental features may change significantly in later releases or be completely removed. Experimental features are not enabled by default and are not supported.
+
+Currently, Notation has the following experimental features:
+
+- Signing, listing and verifying images as OCI image layout
+- Store signatures using artifact manifest (requires registry support). Deprecated in `v1.0.0-rc.5`.
+- Use [Referrers API](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#enabling-the-referrers-api)
```
Comment (yizha1) [inline content/en/docs/concepts/experimental-features.md:14]: Suggest using specific version `v1.1.0-rc2` for the link since `main` is subject to change.

Comment (zr-msft) [reply content/en/docs/concepts/experimental-features.md]: thanks @yizha1 

fixed

File: content/en/docs/concepts/experimental-features.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+description: "Explanation of experimental features"
+type: docs
+weight: 10
+---
+
+Starting in version `v1.0.0-rc.4`, Notation began introducing experimental features. Experimental features are features that are not yet ready for production use but are available for testing and evaluation. Experimental features may change significantly in later releases or be completely removed. Experimental features are not enabled by default and are not supported.
+
+Currently, Notation has the following experimental features:
+
+- Signing, listing and verifying images as OCI image layout
```
Comment (yizha1) [inline content/en/docs/concepts/experimental-features.md:12]: Maybe add a link for user guide, https://notaryproject.dev/docs/how-to/oci-image-layout/

File: content/en/docs/concepts/experimental-features.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- Store signatures using artifact manifest (requires registry support). Deprecated in `v1.0.0-rc.5`.
+- Use [Referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc2/spec.md#enabling-the-referrers-api)
```
Comment (yizha1) [inline content/en/docs/concepts/experimental-features.md:14]: The user guide is not available yet, we can add it later

**Review by yizha1 — COMMENTED** [review]
Thanks @zr-msft one more comment

File: content/en/docs/concepts/experimental-features.md (hunk 3 above)
Comment (zr-msft) [reply content/en/docs/concepts/experimental-features.md]: good call. Fixed

File: content/en/docs/concepts/experimental-features.md (hunk 4 above)
Comment (zr-msft) [reply content/en/docs/concepts/experimental-features.md:14]: Thanks @yizha1 for creating this issue #259 related to this

**Comment by zr-msft** [comment]
Thanks @yizha1  incorporated your feedback

**Review by yizha1 — APPROVED** [review]
LGTM

File: content/en/docs/concepts/experimental-features.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+Currently, Notation has the following experimental features:
+
+- [Signing, listing and verifying images as OCI image layout]({{< ref "/docs/how-to/oci-image-layout" >}})
+- Store signatures using artifact manifest (requires registry support). Deprecated in `v1.0.0-rc.5`.
+- Use [Referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc2/spec.md#enabling-the-referrers-api)
+
+To enable experimental features, set the `NOTATION_EXPERIMENTAL` environment variable to `1`, for example:
+
+```console
+export NOTATION_EXPERIMENTAL=1
```
Comment (FeynmanZhou) [inline content/en/docs/concepts/experimental-features.md:19]: @zr-msft  @yizha1 @shizhMSFT This command is used for Unix-based systems. Do you think it's necessary to add Windows Powershell as an alternative? 

It's not a blocking question so we can merge it as the first version.

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Pull request merged**